webpack/src/Extends/Host/PuppetTab/SubTabs/YamlTab/index.js

```diff
@@ -0,0 +1,60 @@
+import React, { useMemo } from 'react';
+import { useAPI } from 'foremanReact/common/hooks/API/APIHooks';
+import PropTypes from 'prop-types';
+
+import Skeleton from 'react-loading-skeleton';
+import EmptyState from 'foremanReact/components/common/EmptyState/EmptyStatePattern';
+import { STATUS } from 'foremanReact/constants';
+import { EmptyStateIcon } from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import { global_danger_color_200 as dangerColor } from '@patternfly/react-tokens';
+import { translate as __ } from 'foremanReact/common/I18n';
+import './YamlTab.scss';
+
+const YamlTab = ({ hostName }) => {
+  const params = useMemo(
+    () => ({ params: { name: hostName, format: 'yml' } }),
```

Comment:
Why does this need `UseMemo`?

Reply:
So that it can memorize `hostname` instead of recalculating again.

Reply:
But is this calculation that expensive? This will be called once when you open the YAML tab or am I mistaken?